### PR TITLE
Vale: update headings rules

### PR DESCRIPTION
**Pull request by maryamsulemani97**
This PR: 
 - removes `HeadingPunctuation.yml`. We can enable markdownlint's [MD026](https://github.com/markdownlint/markdownlint/blob/master/docs/RULES.md#md026---trailing-punctuation-in-header) instead
 - updates `Headings.yml` so it works for headings like:
   -  `Step 1: Do this`
   - `1.1. do this`
- Minor update to docs to fix heading

part of #1711

### CODE REVIEW

File: .vale/styles/style_guide/Headings.yml [hunk 1]
```diff
@@ -48,4 +49,8 @@ exceptions:
   - App Service Auto-Scale
   - Certbot
   - Bleve & Tantivy
-  - '([0-9a-z]:\s[A-Z])'  #How do get this to throw an error for `Step 1: do this`?
+  - Google Cloud Console
+  - AMIs
+  - '\b\s[0-9]:\s[A-Z]' # Step 1: Do this
+  - '[0-9]\.\s[A-Z]'  # 1. Do this
+  #- '"(.*?)"' #Ignore text in ""
```
Comment (guimachiavelli) [inline .vale/styles/style_guide/Headings.yml:56]:
```suggestion
  #- '"(.*?)"' #Ignore text in ""
  
```

File: .vale/styles/style_guide/Headings.yml [hunk 2]
```diff
@@ -48,4 +49,8 @@ exceptions:
   - App Service Auto-Scale
   - Certbot
   - Bleve & Tantivy
-  - '([0-9a-z]:\s[A-Z])'  #How do get this to throw an error for `Step 1: do this`?
+  - Google Cloud Console
+  - AMIs
+  - '\b\s[0-9]:\s[A-Z]' # Step 1: Do this
+  - '[0-9]\.\s[A-Z]'  # 1. Do this
```
Comment (guimachiavelli) [inline .vale/styles/style_guide/Headings.yml:55]: The two comments are describing the correct capitalisation in these scenarios, right?

**Review by guimachiavelli — DISMISSED** [review]
LGTM 🪲 

Just need to add a blank line at the end of `headings.yml`. Also had a minor question on a couple of comments, just to confirm I am reading it correctly.

File: .vale/styles/style_guide/Headings.yml (hunk 2 above)
Comment (maryamsulemani97) [reply .vale/styles/style_guide/Headings.yml:55]: Yes, the first one is for headings that start with `Step X:` and the second one is for headings that are numbered lists

**Review by dichotommy — APPROVED** [review]
LGTM! 🙆🏻

**Comment by maryamsulemani97** [comment]
bors merge

**Pull request merged**